### PR TITLE
Tweak: Reduce loading controls by removing UI controls [ED-13250]

### PR DESCRIPTION
**Pull request by KingYes**

### CODE REVIEW

File: includes/base/controls-stack.php [hunk 1]
```diff
@@ -2388,7 +2401,9 @@ protected function handle_control_position( array $args, $control_id, $overwrite
 				$args = array_replace_recursive( $target_tab, $args );
 			}
 		} elseif ( empty( $args['section'] ) && ( ! $overwrite || is_wp_error( Plugin::$instance->controls_manager->get_control_from_stack( $this->get_unique_name(), $control_id ) ) ) ) {
-			wp_die( sprintf( '%s::%s: Cannot add a control outside of a section (use `start_controls_section`).', get_called_class(), __FUNCTION__ ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			if ( ! $this->should_optimize_controls() ) {
```
Comment (StyleShit) [inline includes/base/controls-stack.php:2404]: 1. why do you need this condition? is there a case when you don't have a section that is valid in optimized environment?
2. what about cases where someone injects a control after a UI control?

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: includes/base/controls-stack.php [hunk 2]
```diff
@@ -444,7 +431,21 @@ public function add_control( $id, array $args, $options = [] ) {
...
+				Controls_Manager::HEADING,
+				Controls_Manager::BUTTON,
+			];
+
+			if ( ! empty( $args['type'] ) && ! empty( $args['section'] ) && in_array( $args['type'], $ui_controls ) ) {
+				$args = [
+					'type' => $args['type'],
+					'section' => $args['section'],
+				];
+			}
```
Comment (StyleShit) [inline includes/base/controls-stack.php:447]: maybe now you can remove the unset in line 449?

File: includes/base/controls-stack.php [hunk 3]
```diff
@@ -444,7 +431,21 @@ public function add_control( $id, array $args, $options = [] ) {
 			}
 		}
 
-		if ( $should_optimize_controls ) {
+		if ( $this->should_optimize_controls() ) {
+			$ui_controls = [
+				Controls_Manager::RAW_HTML,
+				Controls_Manager::DIVIDER,
+				Controls_Manager::HEADING,
+				Controls_Manager::BUTTON,
```
Comment (StyleShit) [inline includes/base/controls-stack.php:439]: there is a new alert UI control I think, ask @rami-elementor

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: includes/base/controls-stack.php (hunk 3 above)
Comment (rami-elementor) [reply includes/base/controls-stack.php:439]: Yes, two new UI controls. And I see you are missing another existing control

File: includes/base/controls-stack.php [hunk 4]
```diff
@@ -432,6 +432,20 @@ public function add_control( $id, array $args, $options = [] ) {
 		}
 
 		if ( $this->should_optimize_controls() ) {
+			$ui_controls = [
+				Controls_Manager::RAW_HTML,
+				Controls_Manager::DIVIDER,
+				Controls_Manager::HEADING,
+				Controls_Manager::BUTTON,
```
Comment (rami-elementor) [inline includes/base/controls-stack.php:439]:
```suggestion
			$ui_controls = [
				Controls_Manager::RAW_HTML,
				Controls_Manager::DIVIDER,
				Controls_Manager::HEADING,
				Controls_Manager::BUTTON,
				Controls_Manager::ALERT,
				Controls_Manager::NOTICE,
				Controls_Manager::DEPRECATED_NOTICE,
```

**Review by rami-elementor — CHANGES_REQUESTED** [review]

File: includes/base/controls-stack.php (hunk 2 above)
Comment (KingYes) [reply includes/base/controls-stack.php:447]: It's a different.

**Review by StyleShit — APPROVED** [review]

**Review by rami-elementor — APPROVED** [review]

**Pull request merged**